akka-http-core/src/main/scala/akka/http/scaladsl/Http2Shadow.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem, ExtensionId }
+import akka.annotation.InternalApi
+import akka.event.LoggingAdapter
+import akka.http.scaladsl.Http.ServerBinding
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.http.scaladsl.settings.ServerSettings
+import akka.stream.{ ActorMaterializerHelper, Materializer }
+
+import scala.concurrent.Future
+
+/** INTERNAL API: Uses reflection to reach for Http2 support if available or fails with an exception */
+@InternalApi
+private[akka] object Http2Shadow {
+
+  type ShadowHttp2 = AnyRef
+  type ShadowHttp2Ext = AnyRef
+
+  /**
+   * @param httpsContext is required to be an HTTPS context, however we validate this internally,
+   *                     and throw in runtime if it's not. We do this since we support the existing Http() API.
+   */
+  def bindAndHandleAsync(
+    handler:   HttpRequest ⇒ Future[HttpResponse],
+    interface: String, port: Int,
+    httpsContext: ConnectionContext,
+    settings:     ServerSettings,
+    parallelism:  Int,
+    log:          LoggingAdapter)(implicit fm: Materializer): Future[ServerBinding] = {
+
+    val mat = ActorMaterializerHelper.downcast(fm)
+    require(httpsContext.isSecure, "In order to use HTTP/2 you MUST provide an HttpsConnectionContext.")
```

Comment:
Ok, we could force it like that as well, but that would mean if you want to have both an HTTP and an HTTPS/Http2 server you need to have two sets of settings.